logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/SyncConversationsUseCase.kt

```diff
@@ -19,16 +19,39 @@
 package com.wire.kalium.logic.feature.conversation
 
 import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.conversation.Conversation
 import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.SystemMessageBuilder
 import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
 
 /**
  * This use case will sync against the backend the conversations of the current user.
  */
-class SyncConversationsUseCase(private val conversationRepository: ConversationRepository) {
-
-    suspend operator fun invoke(): Either<CoreFailure, Unit> {
-        return conversationRepository.fetchConversations()
-    }
+class SyncConversationsUseCase(
+    private val conversationRepository: ConversationRepository,
+    private val systemMessageBuilder: SystemMessageBuilder
+) {
+    suspend operator fun invoke(): Either<CoreFailure, Unit> =
+        conversationRepository.getConversationIdsByProtocol(Conversation.Protocol.PROTEUS)
+            .flatMap { proteusConversationIds ->
+                conversationRepository.fetchConversations()
```

Comment:
why we here fetching all conversations?

Reply:
It's a step in the slow sync where we fetch all conversations. That was happening before this PR but now I also need to find out if the protocol was changed after re-fetching the conversations because that means that you missed the migration and could have missed messages.

I'm not super happy about having it mixed up with the `SyncConversationsUseCase` but didn't see any other way. If you see any other way I'm very interested.

Reply:
I believe we'd need some not-so-simple mechanism in order to plug in some arbitrary logic pre/post a slow sync step.

Some `SyncStepInterceptor` kind of thing.

I'm with @typfel that this is a good compromise. Not perfect, but looks the best for what we currently have.